### ISSUE TITLE
[Rule Impl] Part 4 style_hover_persistent

### ISSUE THREAD

**Issue by drjoho** [IBMa/equal-access#671]
Follow on to #590 and #614

Write a rule based on detection of the :hover CSS pseudo class. To start, we will look at where :hover triggers another element to display and ensure the displayed element is the next element in the DOM from the triggering element.

- [x] analysis of the failure cases
- [x] discussion on which failure cases to cover since I found 6 additional cases.
- [x] resolution on failure cases of which there are 3
- [x] test cases written (Dev)
- [x] necessary files for the new rule created or edited as needed (Dev)
- [x] logic written in pseudocode (Dev)
- [x] rule logic code (Dev) 
- [x] test rule logic with test cases and add cases as needed (Dev)
- [x] ensure messages are output and unit tests work (Dev)

Documentation at https://ibm.ent.box.com/file/521562780202

**Comment by drjoho** [comment]
The rule is completed. I have some cleanup today. Will move to Review at the end of the day.

**Comment by drjoho** [comment]
Will move to Review and will do some cleanup of logging while the review is taking place.
